src/state/psarstake/hederaHooks.ts

```diff
@@ -196,6 +208,191 @@ export function useDerivativeHederaSarStake(positionId?: BigNumber) {
   );
 }
 
+/**
+ * This hook return the rent value of a position
+ * @param positionId The id of position
+ * Returns rent value in tiny bars
+ */
+function useHederaRent(positionId: string | undefined) {
+  const sarStakingContract = useSarStakingContract();
+  const blockTimestamp = useGetBlockTimestamp();
+
+  const positionState = useSingleCallResult(
+    positionId ? sarStakingContract : undefined,
+    'positions',
+    positionId ? [positionId] : [],
+  );
+
+  const { data: exchangeRate, isLoading: isLoadingRate } = useQuery(
+    'get-hedera-exchange-rate',
+    async () => {
+      const rate = await hederaFn.getExchangeRate();
+      return rate;
+    },
+    {
+      cacheTime: 60 * 1000, // 1 minute
+    },
+  );
+
+  /*
+    rentTime = block.timestamp - position.lastUpdate;
+    rentAmount = rentTime * tinyCentsToTinyBars(500_000_000) / (90 days/s);
+  */
+
+  return useMemo(() => {
+    const isLoading = isLoadingRate || positionState.loading;
+    if (!positionId || !blockTimestamp || isLoading || !exchangeRate || positionState.error || !positionState.valid) {
+      return undefined;
+    }
+
+    try {
+      const tinyBars = hederaFn.tinyCentsToTinyBars('500000000', exchangeRate.current_rate);
+      const lastUpdate = positionState.result?.lastUpdate;
+      const rentTime = Number(blockTimestamp) - lastUpdate;
+      const days = 90 * 24 * 60 * 60;
+      const rentAmount = JSBI.divide(
+        JSBI.multiply(JSBI.BigInt(rentTime.toString()), JSBI.BigInt(tinyBars)),
+        JSBI.BigInt(days.toString()),
+      );
+      return new Fraction('12', '10').multiply(rentAmount).toFixed(0);
+    } catch {
+      return undefined;
+    }
+  }, [positionId, blockTimestamp, exchangeRate, isLoadingRate, positionState, sarStakingContract]);
+}
+
+/**
+ *
+ * @param position Id of a Posttion
+ * @returns Return some utils functions for unstake
+ */
+export function useDerivativeHederaSarUnstake(position: Position | null) {
+  const [typedValue, setTypedValue] = useState('');
+  const [stepIndex, setStepIndex] = useState(0);
+  const [unstakeError, setUnstakeError] = useState<string | null>(null);
+
+  const [attempting, setAttempting] = useState(false);
+  const [hash, setHash] = useState<string | null>(null);
+
+  const { account } = usePangolinWeb3();
+  const chainId = useChainId();
+
+  const { t } = useTranslation();
+  const addTransaction = useTransactionAdder();
+
+  const png = PNG[chainId];
+
+  const sarStakingContract = useSarStakingContract();
+
+  const stakedAmount = new TokenAmount(png, (position?.balance ?? 0).toString());
+
+  const { parsedAmount, error } = useUnstakeParseAmount(typedValue, png, stakedAmount);
+
+  // used for max input button
+  const maxAmountInput = maxAmountSpend(chainId, stakedAmount);
+
+  const wrappedOnDismiss = useCallback(() => {
```

Comment:
I am adding comment here but this is apply to entire hook `useDerivativeHederaSarUnstake`. In this hook and in hook `useDerivativeSarUnstake` there is logic which is not related to any chain, that is general logic. for e.g. `wrappedOnDismiss`, `onUserInput`, `onUserInput`, `onChangePercentage` and many other things are not chain related. only thing which are chain related here is `tinyRent` & `onUnstake`. 

The issue is, whenever we go to new chain, we will just copy paste this logic into new chain hook which will keep creating duplicate code which will be hard to maintain. Please move this common logic to component. Here is an example what i am referring, please check `useStakeCallback` in `src/components/Pools/PangoChef/Stake/index.tsx` and `useHederaPangoChefStakeCallback` in `src/state/ppangoChef/hooks.ts`. 

let me know your thoughts.